railties/lib/rails/commands/server/server_command.rb

```diff
@@ -99,7 +99,7 @@ class ServerCommand < Base # :nodoc:
 
       class_option :port, aliases: "-p", type: :numeric,
         desc: "Runs Rails on the specified port.", banner: :port, default: 3000
-      class_option :binding, aliases: "-b", type: :string, default: "localhost",
```

Comment:
You could add to the description of the option that it defaults to localhost in development and 0.0.0.0 in all other environments, e.g:

``` ruby
class_option :binding, aliases: "-b", type: :string,
  desc: "Binds Rails to the specified IP - defaults to 'localhost' in development and '0.0.0.0' in other environments'.", banner: :IP

```